src/toolkits/style_transfer/style_transfer_model_definition.cpp

```diff
@@ -20,14 +22,24 @@ using CoreML::Specification::NeuralNetworkLayer;
 using turi::neural_net::float_array_map;
 using turi::neural_net::model_spec;
 using turi::neural_net::scalar_weight_initializer;
+using turi::neural_net::xavier_weight_initializer;
 using turi::neural_net::zero_weight_initializer;
 
 using padding_type = model_spec::padding_type;
 
 namespace {
 
+constexpr float LOWER_BOUND = -0.7; 
+constexpr float UPPER_BOUND = 0.7; 
+
 // TODO: refactor code to be more readable with loops
-void define_resnet(model_spec& nn_spec, size_t num_styles) {
+void define_resnet(model_spec& nn_spec, size_t num_styles, int random_seed=0) {
+  std::mt19937 random_engine;
+  std::seed_seq seed_seq{random_seed};
+  random_engine = std::mt19937(seed_seq);
+
+  auto initializer = xavier_weight_initializer(LOWER_BOUND, UPPER_BOUND, &random_engine);
```

Comment:
I'm really confused by this line. First off, it would probably be best not to use `auto` here.

This is what you're calling, right? https://github.com/apple/turicreate/blob/fb1725cffa5cbd188e281626771cb8b6a53b5ae8/src/ml/neural_net/weight_init.cpp#L25

If so, that doesn't look right to me. The first two parameters seem wrong. fan in/out seems different than lower/upper bound. At the very least these are getting covered from float to size_t which does seem right since they aren't whole numbers.

Reply:
Good catch! I was testing it and the stylization didn't work, properly.